packages/victory-core/src/victory-label/victory-label.js

```diff
@@ -110,36 +115,225 @@ const getTransform = (props) => {
+    const prevPaddingProp = getPaddingProp(i - 1, padding);
+
+    return {
+      textHeight,
+      labelSize,
+      y,
+      fontSize: style.fontSize || defaultStyles.fontSize,
+      dy:
+        i && !inline
+          ? previousStyle.fontSize * previousLineHeight +
+            prevPaddingProp.top +
+            prevPaddingProp.bottom
+          : dy - totalLineHeight * 0.5 - (currentStyle.fontSize - capHeightPx)
+    };
+  });
+
+  const backgroundStyleChildren = backgroundStyle.map((bgStyle, i) => {
+    const xCoordinate = getXCoordinate(calculatedProps, textElement[i].labelSize.width);
+    const yCoordinate = textElement.slice(0, i + 1).reduce((prev, curr) => {
+      return prev + curr.dy;
+    }, y);
+    const transform = getTransform(props);
+    const paddingProp = getPaddingProp(i, padding);
+
+    const backgroundProps = {
+      key: `bgKey-${i}`,
+      height: textElement[i].textHeight + paddingProp.top + paddingProp.bottom,
+      style: bgStyle,
+      width: textElement[i].labelSize.width + paddingProp.left + paddingProp.right,
+      transform,
+      x: xCoordinate,
+      y: yCoordinate
+    };
+
+    return React.cloneElement(
+      backgroundComponent,
+      defaults({}, backgroundComponent.props, backgroundProps)
     );
+  });
+
+  return backgroundStyleChildren;
+};
+
+const getBackgroundElement = (props, calculatedProps) => {
+  return Array.isArray(props.backgroundStyle)
+    ? getChildBackgrounds(props, calculatedProps)
+    : getFullBackground(props, calculatedProps);
+};
+
+const calculateSpanDy = (props, calculatedProps, i) => {
+  const { capHeight, style } = props;
+  const { lineHeight } = calculatedProps;
+  const currentStyle = style[i] || style[0];
+  const lastStyle = style[i - 1] || style[0];
+  const previousFontSize = lastStyle.fontSize || defaultStyles.fontSize;
+  const prevCapHeightPx = TextSize.convertLengthToPixels(`${capHeight}em`, lastStyle.fontSize);
+  const previousLineHeight = lineHeight[i - 1] || 1;
+  const fontSize = currentStyle.fontSize || defaultStyles.fontSize;
+  const currentLineHeight = lineHeight[i] || 1;
+  const capHeightPx = TextSize.convertLengthToPixels(`${capHeight}em`, currentStyle.fontSize);
+
+  return (
+    -0.5 * previousFontSize -
+    0.5 * (previousFontSize * previousLineHeight) +
+    previousFontSize * previousLineHeight +
+    0.5 * fontSize +
+    0.5 * fontSize * currentLineHeight -
+    (fontSize - capHeightPx) * 0.5 +
+    (previousFontSize - prevCapHeightPx) * 0.5
+  );
+};
+
+const getTSpanDy = (props, calculatedProps, i) => {
+  const { backgroundStyle, inline } = props;
+  const { padding } = calculatedProps;
+  const paddingProp = getPaddingProp(i, padding);
+  const prevPaddingProp = getPaddingProp(i - 1, padding);
+
+  if (i && !inline) {
+    return backgroundStyle && Array.isArray(backgroundStyle) && backgroundStyle.length > 1
+      ? calculateSpanDy(props, calculatedProps, i) + prevPaddingProp.top + prevPaddingProp.bottom
```

Comment:
I don't know that the background padding should influence the text positioning

Reply:
This is a frustrating one because it has to for the multi-background, but there is a condition there to make sure it's only applied when there are actually multiple background styles

Reply:
No, I think the background padding explicitly _should not_ change the text positioning. I understand that means that when padding is applied, background rects might not exactly line up with one another. But, the alternative is that capHeight / lineHeight are non-deterministic for labels when there is a backgroundPadding prop.

Reply:
<img width="416" alt="Screen Shot 2020-06-15 at 9 44 22 PM" src="https://user-images.githubusercontent.com/3719995/84733158-52d39b00-af52-11ea-8c74-6b66862a0df0.png">

vs

<img width="440" alt="Screen Shot 2020-06-15 at 9 45 08 PM" src="https://user-images.githubusercontent.com/3719995/84733167-58c97c00-af52-11ea-9f57-a3ff924828c8.png">

Reply:
I think it's more correct to maintain the spacing between text elements than to make the background line up, but maybe this is a matter of preference.